tests/api_tests/nodes/test_views.py

```diff
@@ -605,6 +610,83 @@ def test_partial_update_private_project_logged_in_non_contributor(self):
 
+
+class TestNodeDelete(ApiTestCase):
+
+    def setUp(self):
+        ApiTestCase.setUp(self)
+        self.user = UserFactory.build()
+        self.user.set_password('password')
+        self.user.save()
+        self.basic_auth = (self.user.username, 'password')
+
+        self.project = ProjectFactory(creator=self.user, is_public=False)
+        self.private_url = '/{}nodes/{}/'.format(API_BASE, self.project._id)
+
+        self.user_two = UserFactory.build()
+        self.user_two.set_password('password')
+        self.user_two.save()
+        self.basic_auth_two = (self.user_two.username, 'password')
+
+        self.public_project = ProjectFactory(is_public=True, creator=self.user)
+        self.public_url = '/{}nodes/{}/'.format(API_BASE, self.public_project._id)
+
+        self.fake_url = '/{}nodes/{}/'.format(API_BASE, '12345')
+
+    def test_deletes_public_node_logged_out(self):
+        res = self.app.delete(self.public_url, expect_errors=True)
+        # This is 403 instead of 401 because basic authentication is only for unit tests and, in order to keep from
+        # presenting a basic authentication dialog box in the front end. We may change this as we understand CAS
+        # a little better
+        assert_equal(res.status_code, 403)
+
+    def test_deletes_public_node_logged_in(self):
+        res = self.app.delete_json(self.public_url, auth=self.basic_auth_two, expect_errors=True)
+        assert_equal(res.status_code, 403)
+        assert_equal(self.public_project.is_deleted, False)
+
+        res = self.app.delete_json(self.public_url, auth=self.basic_auth, expect_errors=True)
+        assert_equal(res.status_code, 202)
+        assert_equal(self.public_project.is_deleted, False)
+        returned_url = res.json['data']['links']['confirm_delete']
+
+        res = self.app.delete(returned_url, auth=self.basic_auth)
+        assert_equal(res.status_code, 204)
+        assert_equal(self.public_project.is_deleted, True)
+
+    def test_deletes_private_node_logged_out(self):
+        res = self.app.delete(self.private_url, expect_errors=True)
+        # This is 403 instead of 401 because basic authentication is only for unit tests and, in order to keep from
+        # presenting a basic authentication dialog box in the front end. We may change this as we understand CAS
+        # a little better
+        assert_equal(res.status_code, 403)
+
+    def test_deletes_private_node_logged_in_contributor(self):
+        res = self.app.delete(self.private_url, auth=self.basic_auth, expect_errors=True)
+        assert_equal(res.status_code, 202)
+        assert_equal(self.project.is_deleted, False)
+        returned_url = res.json['data']['links']['confirm_delete']
+
+        res = self.app.delete(returned_url, auth=self.basic_auth)
+        assert_equal(res.status_code, 204)
+        assert_equal(self.project.is_deleted, True)
+
+    def test_deletes_private_node_logged_in_non_contributor(self):
```

Comment:
Ooh, we're probably going to need that test for the other write functionality. Not part of this PR, though.